File: JANL.VB/SQL/ISQLCommand.vb
Imports System.Data.SqlClient
Imports System.Threading.Tasks

Namespace SQL

	Public Interface ISQLCommand(Of T)
		ReadOnly Property Parameters As SqlParameterCollection
		Property Timeout As Integer

		Function Execute() As T

		Function Execute(Connection As SqlConnection) As T

		Function Execute(ConnectionString As String) As T

		Function Execute(Transaction As SqlTransaction) As T

		Function ExecuteAsync() As Task(Of T)

		Function ExecuteAsync(Connection As SqlConnection) As Task(Of T)

		Function ExecuteAsync(ConnectionString As String) As Task(Of T)

		Function ExecuteAsync(Transaction As SqlTransaction) As Task(Of T)

	End Interface

End Namespace

File: JANL.CS/SQL/ISQLCommand.cs
﻿using System.Data.SqlClient;
using System.Threading.Tasks;

namespace JANL.SQL
{
	public interface ISQLCommand<T>
	{
		SqlParameterCollection Parameters { get; }

		int Timeout { get; set; }

		T Execute();

		T Execute(SqlConnection Connection);

		T Execute(string ConnectionString);

		T Execute(SqlTransaction Transaction);

		Task<T> ExecuteAsync();

		Task<T> ExecuteAsync(SqlConnection Connection);

		Task<T> ExecuteAsync(string ConnectionString);

		Task<T> ExecuteAsync(SqlTransaction Transaction);
	}
}